ui/raidboss/raidboss_config.js

```diff
@@ -526,24 +583,33 @@ class RaidbossConfigurator {
         try {
           // Can't use ValueOrFunction here as r returns a non-localizable object.
           // FIXME: this hackily replicates some raidboss logic too.
-          let response = (typeof r == 'function') ? r(kFakeData[d], kFakeMatches) : r;
+          let response = r;
+          while (typeof response === 'function')
+            response = r(kFakeData[d], kFakeMatches, trig.output);
           if (!response)
             continue;
-          for (let i = 0; i < keys.length; ++i)
-            evalTrigger(response, keys[i], d);
+
+          if (Object.keys(trig.outputStrings).length === 0) {
+            for (const key of keys)
+              evalTrigger(response, key, d);
+          }
+          break;
         } catch (e) {
           continue;
         }
       }
     }
 
-    for (let i = 0; i < keys.length; ++i) {
-      let key = keys[i];
-      if (!trig[key])
-        continue;
-      for (let d = 0; d < kFakeData.length; ++d) {
-        if (evalTrigger(trig, key, d))
-          break;
+    // Only evaluate fields if there are not outputStrings.
+    // outputStrings will indicate more clearly what the trigger says.
+    if (Object.keys(trig.outputStrings).length === 0) {
```

Comment:
Kind of an odd request, but can these checks:
```js
if (Object.keys(trig.outputStrings).length === 0)
```
Just be:
```js
if (!trig.outputStrings)
```
and get rid of this line above?
```js
trig.outputStrings = trig.outputStrings || {};
```
It feels odd to set yourself up in this way only to check something self-inflicted, unless there's something I'm missing

Reply:
No, I think I was maybe thinking about wanting it to always be defined for the `Object.assign` above it, but I can just handle it being undefined there instead.  Thanks!